scripts/release.ts

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env -S npx ts-node -P tsconfig.scripts.json
+
+import { execSync } from "child_process";
+
+const SV_COMMAND = "yarn run standard-version --infile ./CHANGELOG.md";
+
+const verboseExec = (cmd: string) => {
+  let result;
+
+  try {
+    result = execSync(cmd, { stdio: "pipe" }).toString("utf-8");
+    console.log(`  ✔ ${cmd}`);
+  } catch (error) {
+    throw new Error(`Command \`${cmd}\` failed:\n\n${error}`);
+  }
+
+  return result;
+};
+
+const checkPrerequisites = () => {
+  try {
+    execSync("git --version", {
+      stdio: "ignore",
+    });
+    execSync("hub --version", {
+      stdio: "ignore",
+    });
+  } catch {
+    throw new Error(`
+  Please install missing dependencies:
+    - https://git-scm.com
+    - https://github.com/github/hub
+`);
+  }
+};
+
+const getNewVersion = (): string => {
+  const svResult = execSync(`${SV_COMMAND} --dry-run`, {
+    stdio: "pipe",
+  }).toString("utf-8");
+
+  let version;
+  const VERSION_MATCH = "bumping version in package.json from (.*) to (.*)";
+  const match = svResult.match(VERSION_MATCH);
+
+  if (match) {
+    version = match[2];
+  } else {
+    throw new Error("Could not get version");
+  }
+
+  return version;
+};
+
+const finalizeRelease = () => {
+  const version = process.argv[3];
+  const pullRequestId = process.argv[4];
+
+  if (version === undefined || pullRequestId === undefined) {
+    printWrongArgsMsg();
+    return;
+  }
+
+  console.log(`Finalizing release v${version}:\n`);
+
+  const mergeResult = verboseExec(
+    `hub api -XPUT ` +
+      `"repos/radicle-dev/radicle-upstream/pulls/${pullRequestId}/merge" ` +
+      `--raw-field 'merge_method=squash'`
+  );
+  const releaseCommitSHA = JSON.parse(mergeResult).sha;
+  verboseExec("git checkout master && git pull");
+  verboseExec(`git tag v${version} ${releaseCommitSHA}`);
+  verboseExec(`git push --tags`);
+
+  console.log(`\nRelease v${version} successfully completed! 👏 🎉 🚀\n`);
+};
+
+const printWrongArgsMsg = () =>
+  console.log(`
+  This command should not be run stand-alone.
+  You should run \`yarn release\` and follow the instructions.
+`);
+
+const cutRelease = (version: string): void => {
+  console.log(`\nCutting release v${version}:\n`);
+
+  verboseExec("git checkout master");
+  verboseExec(
+    `git branch release-v${version} && git checkout release-v${version}`
+  );
+  verboseExec(SV_COMMAND);
```

Comment:
If there are typing I’d prefer to use `standard-version` [programatically](https://github.com/conventional-changelog/standard-version#code-usage).

Reply:
I tried to use it programmatically, but there doesn't seem to be a way to capture the `--dry-run` output which I'm using to determine the target version number.

Reply:
5c91f5af2d9d6e41883d108408665e4ba1abe656